src/widget/form/genericchatform.cpp

```diff
@@ -227,9 +227,6 @@ GenericChatForm::GenericChatForm(const Contact* contact, QWidget* parent)
     copyLinkAction = menu.addAction(QIcon(), QString(), this, SLOT(copyLink()));
     menu.addSeparator();
 
-    connect(chatWidget, &ChatLog::customContextMenuRequested, this,
-            &GenericChatForm::onChatContextMenuRequested);
-
     connect(searchForm, &SearchForm::searchInBegin, this, &GenericChatForm::searchInBegin);
```

Comment:
`loadHistoryAction`, `exportChatAction` declare in ChatForm class(in file `chatform.h`), I need to change their enable/disable attribute in it's definition(in file `chatform.cpp`). I have ever implemented to connect `&ChatLog::customContextMenuRequested` signal with `&ChatForm::onChatContextMenuRequested` slot in chatform.cpp. when the `ChatLog::customContextMenuRequested` signal emited, `&GenericChatForm::onChatContextMenuRequested` trigger first then `&ChatForm::onChatContextMenuRequested` trigger. The context menu popup with no enable/disable effect because of carry out sequence.
ChatForm is child class of GenericChatForm. It seems strangely to connect the same signal with child as well as his father.